src/Control Tasks/RockblockControlTask.cpp

```diff
@@ -350,31 +356,66 @@ void RockblockControlTask::dispatch_process_mt_status()
 
 void RockblockControlTask::dispatch_read_message()
 {
-#ifdef VERBOSE_IMUD
     Serial.println("SENT: AT+SBDRBr");
-#endif
-    serial.print("AT+SBDRB\r");
+    sfr::rockblock::serial.print("AT+SBDRB\r");
     transition_to(rockblock_mode_type::process_command);
 }
 
 void RockblockControlTask::dispatch_process_command()
 {
-    if (serial.read() == 'B') {
-        serial.read();
-        serial.read();
-        serial.read();
-        serial.read();
-        serial.read();
-        serial.read();
+    if (sfr::rockblock::serial.read() == 'B') {
+        sfr::rockblock::serial.read();
+        sfr::rockblock::serial.read();
+        sfr::rockblock::serial.read();
+        sfr::rockblock::serial.read();
+        sfr::rockblock::serial.read();
+        sfr::rockblock::serial.read();
+
+        uint32_t opcodes[(int)sfr::rockblock::max_commands_count];
+        uint32_t args_1[(int)sfr::rockblock::max_commands_count];
+        uint32_t args_2[(int)sfr::rockblock::max_commands_count];
+        bool checksum_complete = false;
 
         /*
             Parses up to `max_commands_count` number of commands
             Exits early if end-of-command-upload flags read
         */
         for (int i = 0; i < sfr::rockblock::max_commands_count; i++) {
-            uint8_t look_ahead1 = serial.read(); // Peek
-            uint8_t look_ahead2 = serial.read(); // Peek
+            uint8_t look_ahead1 = sfr::rockblock::serial.read(); // Peek
+            uint8_t look_ahead2 = sfr::rockblock::serial.read(); // Peek
             if (look_ahead1 == constants::rockblock::end_of_command_upload_flag1 && look_ahead2 == constants::rockblock::end_of_command_upload_flag2) {
+                uint8_t transmitted_checksum[constants::rockblock::checksum_len];
+#ifdef VERBOSE_RB
+                Serial.print("Transmitted Checksum: ");
+#endif
+                for (size_t cs = 0; cs < constants::rockblock::checksum_len; ++cs) {
+                    transmitted_checksum[cs] = sfr::rockblock::serial.read();
+                    if (transmitted_checksum[cs] < 0x10)
+                        Serial.print(0, HEX);
+                    Serial.print(transmitted_checksum[cs], HEX);
+                }
+#ifdef VERBOSE_RB
+                Serial.println();
+#endif
+                uint32_t calculated_checksum = 0;
+                for (int j = 0; j < i; j++) {
+                    // Longitudinal redundancy check, kinda
+                    calculated_checksum = calculated_checksum ^ (opcodes[j] ^ args_1[j] ^ args_2[j]);
```

Comment:
It looks like the opcodes is 2 bytes while the args are 4 each. Is that correct? And if so, is there any concern that this would weaken the accuracy of the checksum?